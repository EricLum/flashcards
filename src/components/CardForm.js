import React from 'react'
import PropTypes from 'prop-types'

const CardForm = (props) => {

    return(<div className='genericForm'>
      <form onSubmit={props.submitFunction} onChange={props.handleChange}>
        <div>
          <label>Title: </label>
          <input type='text' name='title' />
        </div>

        <div>
          <label>Question: </label>
          <input type='text' name='question'/>
        </div>

        <div>
          <label>Answer: </label>
          <input type='text' name='answer' />
        </div>

        <input type='submit' />
    </form>
      </div>)
  }

  CardForm.propTypes = {
    onSubmit: PropTypes.func,
    onChange: PropTypes.func,
  }

export default CardForm
