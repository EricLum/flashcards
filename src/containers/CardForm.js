import React, {Component} from 'react'
import CardAdapter from '../adapters/CardAdapter'

class CardForm extends Component {
  constructor(props){
    super(props)

    this.onSubmit =this.onSubmit.bind(this)
    this.handleInput = this.handleInput.bind(this)
  }

  state = {
    title: '',
    question: '',
    answer: '',
    deckId: 0
  }

  componentDidMount(){
    console.log(this.props)
    this.setState({
      deckId: this.props.deckId
    })
  }

  onSubmit(e) {
    e.preventDefault()
    //Add submitting behavior
    CardAdapter.post(this.state)
  }

  handleInput(e){
    this.setState({
      [e.target.name]: e.target.value
    })
  }

  render(){
    return(<form onSubmit={this.onSubmit} onChange={this.handleInput}>

        <div>
          <label>Title: </label>
          <input type='text' name='title' />
        </div>

        <div>
          <label>Question: </label>
          <input type='text' name='question'/>
        </div>

        <div>
          <label>Answer: </label>
          <input type='text' name='answer' />
        </div>

        <input type='submit' />
    </form>)
  }
}

export default CardForm
