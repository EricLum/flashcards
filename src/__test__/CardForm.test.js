import React from 'react';
import Enzyme,{ shallow, mount } from 'enzyme';
import renderer from 'react-test-renderer'
import Adapter from 'enzyme-adapter-react-16'
import CardForm from '../components/CardForm'

Enzyme.configure({adapter: new Adapter()})

describe('CardForm', () => {

  it('should be defined', () => {
    expect(CardForm).toBeDefined()
  })
  it('should render correctly', () => {
    const wrapper = shallow(<CardForm />);
    expect(wrapper).toMatchSnapshot()
  })
  it('should render correctly with RTR', () => {
    const tree = renderer
    .create(<CardForm />).toJSON();
    expect(tree).toMatchSnapshot();
  })
})
