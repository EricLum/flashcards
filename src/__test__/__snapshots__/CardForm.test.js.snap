// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CardForm should render correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <CardForm />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div>
          <label>
            Title: 
          </label>
          <input
            name="title"
            type="text"
          />
        </div>,
        <div>
          <label>
            Question: 
          </label>
          <input
            name="question"
            type="text"
          />
        </div>,
        <div>
          <label>
            Answer: 
          </label>
          <input
            name="answer"
            type="text"
          />
        </div>,
        <input
          type="submit"
        />,
      ],
      "onChange": undefined,
      "onSubmit": undefined,
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <label>
              Title: 
            </label>,
            <input
              name="title"
              type="text"
            />,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Title: ",
            },
            "ref": null,
            "rendered": "Title: ",
            "type": "label",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "name": "title",
              "type": "text",
            },
            "ref": null,
            "rendered": null,
            "type": "input",
          },
        ],
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <label>
              Question: 
            </label>,
            <input
              name="question"
              type="text"
            />,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Question: ",
            },
            "ref": null,
            "rendered": "Question: ",
            "type": "label",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "name": "question",
              "type": "text",
            },
            "ref": null,
            "rendered": null,
            "type": "input",
          },
        ],
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <label>
              Answer: 
            </label>,
            <input
              name="answer"
              type="text"
            />,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Answer: ",
            },
            "ref": null,
            "rendered": "Answer: ",
            "type": "label",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "name": "answer",
              "type": "text",
            },
            "ref": null,
            "rendered": null,
            "type": "input",
          },
        ],
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "type": "submit",
        },
        "ref": null,
        "rendered": null,
        "type": "input",
      },
    ],
    "type": "form",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div>
            <label>
              Title: 
            </label>
            <input
              name="title"
              type="text"
            />
          </div>,
          <div>
            <label>
              Question: 
            </label>
            <input
              name="question"
              type="text"
            />
          </div>,
          <div>
            <label>
              Answer: 
            </label>
            <input
              name="answer"
              type="text"
            />
          </div>,
          <input
            type="submit"
          />,
        ],
        "onChange": undefined,
        "onSubmit": undefined,
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <label>
                Title: 
              </label>,
              <input
                name="title"
                type="text"
              />,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Title: ",
              },
              "ref": null,
              "rendered": "Title: ",
              "type": "label",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "name": "title",
                "type": "text",
              },
              "ref": null,
              "rendered": null,
              "type": "input",
            },
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <label>
                Question: 
              </label>,
              <input
                name="question"
                type="text"
              />,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Question: ",
              },
              "ref": null,
              "rendered": "Question: ",
              "type": "label",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "name": "question",
                "type": "text",
              },
              "ref": null,
              "rendered": null,
              "type": "input",
            },
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <label>
                Answer: 
              </label>,
              <input
                name="answer"
                type="text"
              />,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Answer: ",
              },
              "ref": null,
              "rendered": "Answer: ",
              "type": "label",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "name": "answer",
                "type": "text",
              },
              "ref": null,
              "rendered": null,
              "type": "input",
            },
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "type": "submit",
          },
          "ref": null,
          "rendered": null,
          "type": "input",
        },
      ],
      "type": "form",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`CardForm should render correctly with RTR 1`] = `
<form
  onChange={undefined}
  onSubmit={undefined}
>
  <div>
    <label>
      Title: 
    </label>
    <input
      name="title"
      type="text"
    />
  </div>
  <div>
    <label>
      Question: 
    </label>
    <input
      name="question"
      type="text"
    />
  </div>
  <div>
    <label>
      Answer: 
    </label>
    <input
      name="answer"
      type="text"
    />
  </div>
  <input
    type="submit"
  />
</form>
`;
